lib/ansible/module_utils/redfish_utils.py

```diff
@@ -455,6 +474,87 @@ def get_disk_inventory(self, systems_uri):
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        if 'SimpleStorage' not in data and 'Storage' not in data:
+            return {'ret': False, 'msg': "SimpleStorage and Storage resource \
+                     not found"}
+
+        if 'Storage' in data:
+            # Get a list of all storage controllers and build respective URIs
+            storage_uri = data[u'Storage'][u'@odata.id']
+            response = self.get_request(self.root_uri + storage_uri)
+            if response['ret'] is False:
+                return response
+            result['ret'] = True
+            data = response['data']
+
+            if data[u'Members']:
+                for controller in data[u'Members']:
+                    controller_list.append(controller[u'@odata.id'])
+                for c in controller_list:
+                    uri = self.root_uri + c
+                    response = self.get_request(uri)
+                    if response['ret'] is False:
+                        return response
+                    data = response['data']
+
+                    if 'Volumes' in data:
+                        # Get a list of all volumes and build respective URIs
+                        volumes_uri = data[u'Volumes'][u'@odata.id']
+                        response = self.get_request(self.root_uri + volumes_uri)
+                        data = response['data']
+
+                        if data[u'Members']:
+                            for volume in data[u'Members']:
+                                volume_list.append(volume[u'@odata.id'])
+                            for v in volume_list:
+                                uri = self.root_uri + v
+                                response = self.get_request(uri)
+                                if response['ret'] is False:
+                                    return response
+                                data = response['data']
+
+                                volume_result = {}
+                                for property in properties:
+                                    if property in data:
+                                        if data[property] is not None:
+                                            volume_result[property] = data[property]
+
+                                # Get related Drives Id
+                                drive_id_list = []
+                                if 'Links' in data:
+                                    if 'Drives' in data[u'Links']:
+                                        for link in data[u'Links'][u'Drives']:
+                                            drive_id_link = link[u'@odata.id']
+                                            drive_id = drive_id_link.split("/")[-1]
+                                            drive_id_list.append({'Id': drive_id})
+                                        volume_result['Linked_disk'] = drive_id_list
```

Comment:
Let's use the Redfish "drive" terminology here instead of "disk".

```suggestion
                                        volume_result['Linked_drives'] = drive_id_list
```

Reply:
Many thanks, billdodd, I fixed the code as you suggested.